[improve][pulsar-client-tools][ISSUE-16082] - Updated set retention time description and added test cases including default time

mans2singh opened the pull request:
Fixes #16082 

### Motivation

* The help option for set retention indicates `-t` option is `Retention time in minutes` but if no time unit is specified, it is assumed to be seconds.

### Modifications

* Updated description for time option and added unit test cases with default value (without time unit suffix) and with seconds time unit.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
  - Added unit tests for to `PulsarAdminToolTest` with and without time unit suffix.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
This is a change in argument description.
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
@Anonymitaet  - 

Can you please review this PR regarding issue (https://github.com/apache/pulsar/issues/16082)  ?  It clarifies that the default time unit (without time unit) is seconds. I have added supporting unit test cases with and without time unit.  

From my understanding the failing unit tests (TEST-org.apache.pulsar.broker.service.RackAwareTest.xml) is not related this PR or it's tests.  

Please let me know if there is anything else required.

Thanks for your time.

Anonymitaet commented:
@horizonzy 
could you please review this PR from the technical perspective? Thank you!

mans2singh commented:
@Anonymitaet, @horizonzy - Please let me know if you have any feedback on this PR.  Thanks

horizonzy commented:
> @Anonymitaet, @horizonzy - Please let me know if you have any feedback on this PR. Thanks

Fine.

horizonzy reviewed (commented):
Could you add default unit for param `--size` in the docs and test for it?

And there are another two SetRetention class.
1.https://github.com/apache/pulsar/blob/42fe0603518be7db7a14802eb4274b6ea22b0c9a/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1815

2.https://github.com/apache/pulsar/blob/42fe0603518be7db7a14802eb4274b6ea22b0c9a/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java#L464

mans2singh commented:
Hi @horizonzy  - 

Thanks for your feedback.  

I've updated description for size parameter and added test cases as you had recommended.

Please let me know if you have any additional advice.

Thanks again for your time.

mans2singh commented:
/pulsarbot run-failure-checks

horizonzy commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:
```diff
@@ -741,8 +741,13 @@ private class SetRetention extends CliCommand {
         @Parameter(names = { "--time",
-                "-t" }, description = "Retention time in minutes (or minutes, hours,days,weeks eg: 100m, 3h, 2d, 5w). "
-                        + "0 means no retention and -1 means infinite time retention", required = true)
+                "-t" }, description = "Retention time with optional time unit suffix "
```
This file retention size doc miss update.

mans2singh replied:
I missed that one, I've updated it.

horizonzy commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:
```diff
@@ -1812,11 +1812,18 @@ private class SetRetention extends CliCommand {
 
-        @Parameter(names = { "--size", "-s" }, description = "Retention size limit (eg: 10M, 16G, 3T). "
+        @Parameter(names = { "--size", "-s" }, description = "Retention size limit with optional size unit suffix "
+                + "eg: 4096, 10M, 16G, 3T.  The size unit suffix character can be k/K, m/M, g/G, or t/G.  "
```
t/G -> t/T

mans2singh replied:
Thanks for catching my error.  Corrected.

horizonzy commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java:
```diff
@@ -466,11 +466,18 @@ private class SetRetention extends CliCommand {
-                + "0 means no retention and -1 means infinite time retention", required = true)
+                "-t" }, description = "Retention time with optional time unit suffix "
+                + "eg: 100m, 3h, 2d, 5w. "
+                + "The suffix character can be s for seconds, m for minutes, h for hours, "
```
> "The suffix character can be s for seconds, m for minutes, h for hours, d for days, w for weeks or y for years. "  

I think it is not necessary.

mans2singh replied:
Removed the extra line from description.

horizonzy replied:
Still in `CmdTopicPolicies`

mans2singh replied:
Updated based on recommendations.  Thanks

mans2singh commented:
@horizonzy  - Thanks for your thorough feedback.  I've updated the code based on your recommendations.

horizonzy reviewed (approved):
LGTM

horizonzy commented:
/pulsarbot run-failure-checks

horizonzy commented:
@Anonymitaet ping

Anonymitaet commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java:
```diff
@@ -466,11 +466,16 @@ private class SetRetention extends CliCommand {
-                "-t" }, description = "Retention time in minutes (or minutes, hours,days,weeks eg: 100m, 3h, 2d, 5w). "
-                + "0 means no retention and -1 means infinite time retention", required = true)
+                "-t" }, description = "Retention time with optional time unit suffix "
+                + "eg: 100m, 3h, 2d, 5w. "
```
```suggestion
                + "For example, 100m, 3h, 2d, 5w. "
```

mans2singh replied:
Sounds good.  Thanks

Anonymitaet commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java:
```diff
@@ -466,11 +466,16 @@ private class SetRetention extends CliCommand {
-                + "0 means no retention and -1 means infinite time retention", required = true)
+                "-t" }, description = "Retention time with optional time unit suffix "
+                + "eg: 100m, 3h, 2d, 5w. "
+                + "If no time unit is specified the default is seconds, eg: "
```
```suggestion
                + "If the time unit is not specified, the default unit is second. For example, "
```

mans2singh replied:
Updated.

Anonymitaet commented on pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopicPolicies.java:
```diff
@@ -466,11 +466,16 @@ private class SetRetention extends CliCommand {
+                "-t" }, description = "Retention time with optional time unit suffix "
+                + "eg: 100m, 3h, 2d, 5w. "
+                + "If no time unit is specified the default is seconds, eg: "
+                + "-t 120 will set retention to 2 minutes. "
```
```suggestion
                + "-t 120 sets retention to 2 minutes. "
```
Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true.
https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit?pli=1#bookmark=id.e8uqh1awkcnp

mans2singh replied:
Updated.  Thanks

mans2singh commented:
@Anonymitaet - 

I have updated the description based on your recommendations.  Please let me know if there is anything else required.  

Thanks again for your advice/time.

Anonymitaet commented:
Thanks for your contribution! LGTM from the technical writing perspective 😊

mans2singh commented:
@Anonymitaet  - It looks like PR is pending your review feedback (`1 pending reviewer` above).  Please let me know if there is anything I can address.  Thanks again for all your advice.

Anonymitaet reviewed (approved):
Approve from a technical writing perspective.

mans2singh commented:
@Anonymitaet @horizonzy  - Please let me know if there is anything required for this PR to be merged to master. Thanks

Anonymitaet commented:
/pulsarbot run-failure-checks

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
@Anonymitaet @horizonzy  - The failing tests (Pulsar CI/CI - Unit Broker - Flaky, Pulsar CI/ CI - System Pulsar Connectors,  CI - CPP, Python Tests / System-PULSAR_CONNECTORS_THREAD, CI - CPP, Python Tests / Unit-BROKER_FLAKY Tests) are not related this PR and are still failing despite invoking `pulsarbot run-failure-checks`.  

Please let me know if there is anything required from my side to merge this to master.  

Thanks again for your time and advice.

Anonymitaet commented:
@mans2singh your PR can not be merged since some required tests failed. Can you get all required tests passed?

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
/pulsarbot run-failure-checks

mans2singh commented:
Hi @Anonymitaet - 

All checks are passing now.  

Please let me know if there is anything else required.  

Thanks again for your guidance.

mans2singh commented:
Thanks @Anonymitaet @horizonzy for your help.
